catalyst/core/runner.py

```diff
@@ -826,6 +826,8 @@ def _run_loader(self, loader: DataLoader) -> None:
         Args:
             loader (DataLoader): dataloader to iterate
         """
+        assert loader, f"DataLoader with name {self.loader_name} is empty."
```

Comment:
what do you think about copying this assert to `run_epoch` too?
It looks easier to check them instantly on epoch start, rather than wait until (maybe some empty) loader run.

Reply:
nevertheless, this assert is valid too, because we could change the loaders with the callbacks.
btw, why non't you like to use `len(loader) > 0)`? I looks much easier to understand, isn't it?

Reply:
I think that `_run_loader` is better because we can change the loaders via callbacks, as you said.

As for `len(loader) > 0` -- ok, I will fix it.

Reply:
Yup, we definitely need that check on `_run_loader`, but why should not we also add some kind of pre-check on `_run_epoch`? what do you think about it?

Reply:
Okay, done